### PR TITLE
pkg/reloader: use watchInterval timeout for initial apply

### PR DESCRIPTION
**Pull request by captncraig**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

I am using pkg/reloader indirectly through [prometheus-operator/prometheus-config-reloader](https://github.com/prometheus-operator/prometheus-operator/tree/main/cmd/prometheus-config-reloader) to watch a prometheus instance. The reloader is a sidecar container to a prometheus pod.

The problem I am having is that I get in a situation where I have a bad prometheus config, the prometheus instance starts a crash loop, and it never gets to the point of making a /-/reload endpoint availible. The reloader in [pkg/reloader](https://github.com/prometheus-operator/prometheus-operator/tree/main/cmd/prometheus-config-reloader) does an initial apply on startup with no timeout specified. This creates a deadlock, where the reloader does not pick up future changes to the underlying config map (which fixes the crash loop), because it is trying to reload the pod that is crash-looping. 

This change simply applies the `watchInterval` as a timeout value passed via context to `apply`, just as it is in every other invocation.

The effect is that the initial apply will eventually time out, and `Watch` will proceed to watch for changes and attempt to reload the target as usual. 

## Verification

I have built an image for prometheus-operator/prometheus-config-reloader that uses this branch, and confirmed in my case that the reloader is able to timeout, and proceed watching and updating as expected.

### CODE REVIEW

File: pkg/reloader/reloader.go [hunk 1]
```diff
@@ -202,13 +202,14 @@ func (r *Reloader) Watch(ctx context.Context) error {
 	}
 
 	defer runutil.CloseWithLogOnErr(r.logger, r.watcher, "config watcher close")
+	applyCtx, applyCancel := context.WithTimeout(ctx, r.watchInterval)
```
Comment (fpetkovski) [inline pkg/reloader/reloader.go:205]: For readability maybe we can use different variable names, `initialSyncCtx` and `initialSyncCancel`.

**Review by fpetkovski — COMMENTED** [review]
Thanks, just one nit from my side.

**Review by fpetkovski — APPROVED** [review]
Ty!

**Review by GiedriusS — DISMISSED** [review]

**Comment by GiedriusS** [comment]
Could you please resolve the conflict? And let's merge this

**Review by GiedriusS — APPROVED** [review]

**Review by matej-g — APPROVED** [review]

**Pull request merged**